pkgs/native_toolchain_c/lib/src/native_toolchain/msvc.dart

```diff
@@ -63,9 +63,15 @@ Tool vcvars(ToolInstance toolInstance) {
   final tool = toolInstance.tool;
   assert(tool == cl || tool == link || tool == lib);
   final vcDir = toolInstance.uri.resolve('../../../../../../');
-  final fileName = toolInstance.uri.toFilePath().contains('x86')
-      ? 'vcvars32.bat'
-      : 'vcvars64.bat';
+  var fileName = 'vcvars64.bat';
+  if (toolInstance.uri.toFilePath().contains('\\x86\\')) {
+    fileName = 'vcvars32.bat';
+  } else if (toolInstance.uri.toFilePath().contains('\\arm64\\')) {
+    // vcvarsarm64 only works on native windows-arm64. In case of cross
```

Comment:
I don't understand this comment.

When should we use which vcvars file? In all 4 combinations?

1. x64 host x64 target: vcvars64.bat
2. arm64 host, x64 target: ?
3. x64 host, arm64 target: ?
4. arm64 host, arm64 target: ?

Reply:
See comment on next discussion (`vcvarsamd64_arm64.bat`) for more details.

1.  x64 host x64 target: vcvars64.bat
2.  arm64 host, x64 target:  vcvars64.bat
3.  x64 host, arm64 target:  vcvarsamd64_arm64.bat (cross compilation)
4.  arm64 host, arm64 target: vcvarsamd64_arm64.bat (cross compilation, to avoid a special case)

We could optimize case 4, to use native toolchain, but cross compilation works well and it avoids having `if (windows-arm64) {...}` code. If you insist on this, I would prefer to introduce this in another PR.

Reply:
Landing something that works SGTM.

Please file an issue on this repo that we should look into this and leave a TODO in the code with a reference to the issue.

Reply:
https://github.com/dart-lang/native/issues/170
I'll add reference in a comment.